crates/oxc_linter/src/rules/eslint/curly.rs

```diff
@@ -0,0 +1,1904 @@
+                _ => None,
+            };
+            let body_start = body.span().start;
+            let stmt_start = stmt.map_or(body_start, |stmt| stmt.span().start);
+            let comments = ctx.comments_range(body_start..stmt_start - 1);
+
+            stmt.map_or(true, |stmt| !is_one_liner(stmt, ctx) || comments.count() > 0)
+        } else {
+            !is_one_liner(body, ctx)
+        })
+    } else {
+        Some(true)
+    };
+
+    should_block
+}
+
+fn report_if_needed<'a>(
+    ctx: &LintContext<'a>,
+    body: &Statement<'a>,
+    keyword: &str,
+    has_braces: bool,
+    should_have_braces: Option<bool>,
+) {
+    let Some(should_have_braces) = should_have_braces else {
+        return;
+    };
+    if should_have_braces == has_braces {
+        return;
+    }
+
+    ctx.diagnostic_with_fix(curly_diagnostic(body.span(), keyword, should_have_braces), |fixer| {
+        if should_have_braces {
+            let fixed = format!("{{{}}}", ctx.source_range(body.span()));
+            fixer.replace(body.span(), fixed)
+        } else {
+            let needs_preceding_space = ctx
+                .nodes()
+                .parent_node(get_node_by_statement(body, ctx).id())
+                .map_or(false, |parent| matches!(parent.kind(), AstKind::DoWhileStatement(_)));
+            let mut fixed = ctx.source_range(body.span()).to_string();
+
+            if let Some(stripped) = fixed.strip_prefix(|c: char| c.is_whitespace() || c == '{') {
+                fixed = stripped.to_string();
+            }
+
+            if let Some(stripped) = fixed.strip_suffix(|c: char| c.is_whitespace() || c == '}') {
+                fixed = stripped.to_string();
+            } else if fixed.ends_with('}') {
+                fixed.pop();
+            }
+
+            if needs_preceding_space {
+                fixed.insert(0, ' ');
+            }
+            fixer.replace(body.span(), fixed)
+        }
+    });
+}
+
+fn is_collapsed_one_liner(node: &Statement, ctx: &LintContext) -> bool {
+    let node = get_node_by_statement(node, ctx);
+    let span = node.span();
+    let node_string = ctx.source_range(span);
+
+    let trimmed_len =
+        u32::try_from(node_string.trim_end_matches(|c: char| c.is_whitespace() || c == ';').len())
+            .expect("Failed to convert to u32");
+
+    let before_node_span = get_token_before(node, ctx).map_or_else(
+        || {
+            ctx.nodes()
+                .parent_node(node.id())
+                .filter(|parent| parent.span().start < span.start)
+                .map_or(Span::new(0, 0), |parent| {
+                    Span::new(parent.span().start, parent.span().start)
+                })
+        },
+        oxc_span::GetSpan::span,
+    );
+
+    let text = ctx.source_range(Span::new(
+        before_node_span.end + 1,
+        span.end
+            - (u32::try_from(node_string.len()).expect("Failed to convert to u32") - trimmed_len),
```

Comment:
No need to use `try_from` and expect here, it's OK to assume files are not larger than 4GB, it's a general assumption we make. You can use `as u32` and add the annotation `#[allow(clippy::cast_possible_truncation)]`

Reply:
OK! updated: https://github.com/oxc-project/oxc/pull/8123/commits/ea76f7d3b52ff30ac56d8a9186a6b00e2374de04